website/hw/hw6.rst

```diff
@@ -35,406 +43,480 @@ repository, and run some tests. These actions are typically referred to
-   GCC and one with clang), and make sure that the clang one is actually
-   using clang (by looking at the output of the Make commands in the
-   “Job log”). Take the URL of the build you just produced, and paste it
-   into Gradescope (under “Task 3: GCC vs clang”). You must also explain
-   what you did to ensure the clang job used clang.
-
-   Note: If you end up with more than two jobs, make sure you’ve removed
-   any other option that would result in additional jobs being launched.
-
--  [10 + 10 points] Using matrix expansion options *only*, generate a
-   build with four configurations:
-
-   -  GCC on Ubuntu 18.04
-   -  clang on Ubuntu 18.04
-   -  GCC on Mac OS X
-   -  clang on Mac OS X
-
-   You will get 10 points just for producing the build with the correct
-   four jobs. To get the remaining 10 points, you will need to modify
-   your ``.travis.yml`` file further to get the code to build correctly
-   in a Mac OS X environment.
-
-   Hint: as you’ll discover fairly quickly, Mac OS X doesn’t include the
-   ``apt`` command used in Ubuntu systems, and which we rely on to
-   install the Criterion library. Installing *dependencies* can be
-   tricky when combining multiple different environment, so you may
-   focus on the part of the Travis CI documentation that focuses on
-   that.
-
-   Take the URL of the build you just produced, and paste it into
-   Gradescope (under “Task 3: Ubuntu and Mac OS X”). If you got the code
-   to build in a Mac OS X environment, you must also explain how you
-   accomplished this.
-
--  [10 points] It turns out that, in a Mac environment, running ``gcc``
-   just results in a call to clang. So, it seems a bit redundant to run
-   the two Mac jobs in the previous build (which will both end up using
-   clang). Figure out a way to *exclude* the Mac+GCC job from the build.
-   Note: You will not get credit if you simply write out the three job
-   configurations. You must instead continue to use matrix expansion,
-   but figure out a way to exclude one specific configuration.
-
-   Take the URL of the build you just produced, and paste it into
-   Gradescope (under “Task 3: Excluding a job”).
-
-Submitting your lab
--------------------
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        compiler: [gcc, clang]
+
+This would run six different configurations. However, let's say
+we're only interested in testing our code with both GCC and clang
+in the latest version of Ubuntu (for prior versions, we'll just
+test it with GCC). So, for avoidance of doubt, you would end
+up with four jobs:
+
+- Ubuntu 16.04 with GCC
+- Ubuntu 18.04 with GCC
+- Ubuntu 20.04 with GCC
+- Ubuntu 20.04 with clang
+
+Figure out a way to run only the above jobs. For full credit,
+you must do so in a way that does not involve writing out the configuration
+for four separate jobs. Also, take into account that, depending on
```

Comment:
It might help to be more specific about what you mean by "writing out the configuration for four separate jobs". I'm not sure exactly what more could be said, though.